lib/bolt/applicator.rb

```diff
@@ -30,23 +29,16 @@ def initialize(inventory, executor, modulepath, plugin_dirs, project,
 
       @pool = Concurrent::ThreadPoolExecutor.new(max_threads: max_compiles)
       @logger = Logging.logger[self]
-      @plugin_tarball = Concurrent::Delay.new do
-        build_plugin_tarball do |mod|
-          search_dirs = []
-          search_dirs << mod.plugins if mod.plugins?
-          search_dirs << mod.pluginfacts if mod.pluginfacts?
-          search_dirs << mod.files if mod.files?
-          type_files = "#{mod.path}/types"
-          search_dirs << type_files if File.exist?(type_files)
-          search_dirs
-        end
-      end
     end
 
     private def libexec
       @libexec ||= File.join(Gem::Specification.find_by_name('bolt').gem_dir, 'libexec')
     end
 
+    def selected_module?(mod)
+      @selected_modules.any? { |m| mod =~ /#{m}/ }
```

Comment:
```suggestion
      @selected_modules.include?(mod)
```

This will more accurately select the exact module, not ones that include the modulename. For example if I have module `nginx` and call `selected_module?('nginx-config')` this will return true.

Reply:
The idea was to have an array of regular expressions, to make it easy to select multiple modules. So I put it there explicitly. For example, the augueas provides can be added like this:

```
['augeasproviders_.*']
```

Personally I prefer this above specifying all modules individually.